Cannot generate schema for response having a collection of enums (Groovy)

### Expected Behavior

The OpenAPI schema should be generated when a response class contains an attribute that is a collection of enum elements.

### Actual Behaviour

The schema generation fails with the following error when a response class contains an attribute that is a collection of enum elements (at least in Groovy).

```
> Task :compileGroovy FAILED
1 actionable task: 1 executed

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileGroovy'.
> BUG! exception in phase 'semantic analysis' in source unit '/home/user/mn-openapi-enums/src/main/groovy/mn/openapi/enums/controller/DogController.groovy' Cannot invoke "io.swagger.v3.oas.models.media.Schema.setName(String)" because "schema" is null
```

### Steps To Reproduce

1. Download the sample app.
2. Look at the `Dog` class, which contains a `Set<Trait> traits` attribute, being `Trait` an `enum`.
3. Look at the `DogController` class, which contains a method returning a `Dog` instance.
4. Run the `compileGroovy` task.
5. The schema generation fails.

### Environment Information

- Operating System: Linux
- JDK version: Java 17

### Example Application

https://github.com/tcrespog/mn-openapi-enums

### Version

3.9.4

--- a/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java
+++ b/openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java
@@ -792,6 +792,11 @@ abstract class AbstractOpenApiVisitor {
         Boolean isIterable = null;
 
         ClassElement componentType = type != null ? type.getFirstTypeArgument().orElse(null) : null;
+        if (componentType != null && !componentType.isIterable() && !componentType.isArray()) {
+            // bug in micronaut-core, see "test build OpenAPI for set of enums"
+            // type.getFirstTypeArgument() return ClassElement, but must be EnumElement
+            componentType = context.getClassElement(componentType.getName()).orElse(componentType);
+        }
         if (type instanceof WildcardElement) {
             WildcardElement wildcardEl = (WildcardElement) type;
             type = CollectionUtils.isNotEmpty(wildcardEl.getUpperBounds()) ? wildcardEl.getUpperBounds().get(0) : null;

--- a/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiPojoControllerGroovySpec.groovy
+++ b/openapi/src/test/groovy/io/micronaut/openapi/visitor/OpenApiPojoControllerGroovySpec.groovy
@@ -88,4 +88,83 @@ class MyBean {}
         !schema.required.contains('battingAverage')
     }
 
+    @Issue("https://github.com/micronaut-projects/micronaut-openapi/issues/1135")
+    void "test build OpenAPI for set of enums"() {
+        given: "An API definition"
+        when:
+        buildBeanDefinition('test.MyBean', '''
+package test
+
+import groovy.transform.CompileStatic
+import io.micronaut.core.annotation.Nullable
+import io.micronaut.http.HttpResponse
+import io.micronaut.http.annotation.Controller
+import io.micronaut.http.annotation.Get
+import io.swagger.v3.oas.annotations.media.Schema
+
+@CompileStatic
+@Controller
+class DogController {
+
+    @Get('/dog')
+    HttpResponse<Dog> get() {
+        Dog dog = new Dog(
+                name: 'Rex', breed: Breed.GERMAN_SHEPHERD,
+                traits: EnumSet.of(Trait.QUICKNESS, Trait.COURAGE, Trait.LOYALTY)
+        )
+
+        return HttpResponse.ok(dog)
+    }
+}
+
+@CompileStatic
+class Dog {
+
+    String name
+    Breed breed
+    Set<Trait> traits
+}
+
+@CompileStatic
+enum Breed {
+
+    GERMAN_SHEPHERD,
+    BULLDOG,
+    GOLDEN_RETRIEVER
+}
+
+@CompileStatic
+enum Trait {
+
+    QUICKNESS,
+    COURAGE,
+    LOYALTY
+}
+
+@jakarta.inject.Singleton
+class MyBean {}
+''')
+        then: "the state is correct"
+        Utils.testReference != null
+
+        when: "The OpenAPI is retrieved"
+        OpenAPI openAPI = Utils.testReference
+        Schema breedSchema = openAPI.components.schemas['Breed']
+        Schema dogSchema = openAPI.components.schemas['Dog']
+        Schema traitSchema = openAPI.components.schemas['Trait']
+
+        then: "the components are valid"
+        breedSchema
+        dogSchema
+        traitSchema
+
+        dogSchema.properties.traits
+        dogSchema.properties.traits.type == 'array'
+        dogSchema.properties.traits.items.$ref == '#/components/schemas/Trait'
+
+        traitSchema.type == 'string'
+        traitSchema.enum.get(0) == 'QUICKNESS'
+        traitSchema.enum.get(1) == 'COURAGE'
+        traitSchema.enum.get(2) == 'LOYALTY'
+    }
 }

